tests/dataloader_tests.cpp

```diff
@@ -86,4 +86,156 @@ BOOST_AUTO_TEST_CASE(MNISTDataLoaderTest)
+    totalBoundingBoxes += dataloader.TrainLabels()(0, i).n_elem / 5;
+  }
+
+  // Rows will be equal to shape image depth * image width * image height.
+  BOOST_REQUIRE_EQUAL(dataloader.TrainFeatures().n_rows, 64 * 64 * 3);
+  // There are 6 images in training set.
+  BOOST_REQUIRE_EQUAL(dataloader.TrainFeatures().n_cols, 6);
+
+  // There is 1 image in validation set.
+  BOOST_REQUIRE_EQUAL(dataloader.ValidLabels().n_cols, 1);
+  // They correspond to class name, x1, y1, x2, y2.
+  BOOST_REQUIRE_EQUAL(dataloader.ValidLabels().n_rows, 1);
+
+  // Each bounding box contains 5 elements corresponding to class label
+  // and 4 elements corresponding to bounding box coordinates.
+  for (size_t i = 0; i < dataloader.ValidLabels().n_cols; i++)
+  {
+    BOOST_REQUIRE_EQUAL(dataloader.ValidLabels()(0, i).n_elem % 5, 0);
+    totalBoundingBoxes += dataloader.ValidLabels()(0, i).n_elem / 5;
+  }
+
+  // There are 15 objects in the dataset.
+  BOOST_REQUIRE_EQUAL(totalBoundingBoxes, 15);
+
+  // Rows will be equal to shape image depth * image width * image height.
+  BOOST_REQUIRE_EQUAL(dataloader.ValidFeatures().n_rows, 64 * 64 * 3);
+  // There are total 15 objects in the images.
+  BOOST_REQUIRE_EQUAL(dataloader.ValidFeatures().n_cols, 1);
+
+  // Boundary check.
+  validRatio = 1.0;
+  dataloader.LoadObjectDetectionDataset(basePath + annotaionPath,
+      basePath + imagesPath, classes, validRatio, shuffle, augmentation);
+
+  // There are total 7 images in the set.
+  BOOST_REQUIRE_EQUAL(dataloader.ValidLabels().n_cols, 7);
+  // They correspond to class name, x1, y1, x2, y2.
+  BOOST_REQUIRE_EQUAL(dataloader.ValidLabels().n_rows, 1);
+
+  // Rows will be equal to shape image depth * image width * image height.
+  BOOST_REQUIRE_EQUAL(dataloader.ValidFeatures().n_rows, 64 * 64 * 3);
+  // There are total 15 objects in the images.
+  BOOST_REQUIRE_EQUAL(dataloader.ValidFeatures().n_cols, 7);
+}
+
+/**
+ * Simple Test for object detection dataloader.
+ */
+BOOST_AUTO_TEST_CASE(ObjectDetectionDataLoaderMatTypeTest)
+{
+  DataLoader<> dataloader;
+  Utils::ExtractFiles("./../data/PASCAL-VOC-Test.tar.gz", "./../data/");
+
+  // Set paths for dataset.
+  std::string basePath = "./../data/PASCAL-VOC-Test/";
+  std::string annotaionPath = "Uniform_Annotation/";
+  std::string imagesPath = "Images/";
+  double validRatio = 0.2;
+  bool shuffle = true;
+
+  // Classes in the dataset.
+  std::vector<std::string> classes = {"background", "aeroplane", "bicycle",
+      "bird", "boat", "bottle", "bus", "car", "cat", "chair", "cow",
+      "diningtable", "dog", "horse", "motorbike", "person", "pottedplant",
+      "sheep", "sofa", "train", "tvmonitor"};
+
+  // Resize the image to 64 x 64.
+  std::vector<std::string> augmentation = {"resize (64, 64)"};
+  dataloader.LoadObjectDetectionDataset(basePath + annotaionPath,
+      basePath + imagesPath, classes, validRatio, shuffle, augmentation);
+
+  // There are total 15 objects in images.
```

Comment:
Because some images have multiple objects in the dataset, let's use the term `images` instead of `objects`.
```suggestion
  // There are 6 images in the training set.
```